core/templates/services/request-interceptor.service.spec.ts

```diff
@@ -197,4 +197,48 @@ describe('Request Interceptor Service', () => {
           {} as HttpHandler);
     }).toThrowError('Cannot supply params with value "None" or "null".');
   });
+
+  it('should throw error if null param in DELETE request', () => {
+    expect(() => {
+      requestInterceptor.intercept(
+        new HttpRequest(
+          'DELETE',
+          'url',
```

Comment:
It looks like the `url` parameter refers to the _endpoint_ URL without any query params and the HTTP library will not parse those params correctly. That seems more related to the intended use of the library classes/methods though.

Reply:
But do we currently do any calls to GET/DELETE in our codebase with the URL containing query params? If so then this seems like a case we should guard against somehow...

Reply:
I see, well I would think that the corresponding API tests would catch those errors, but I suppose we could raise an error if we detect query params in the endpoint URL itself.

Reply:
I don't think we throw an error for those cases actually, or consider that an error -- that's why I'm suggesting looking into this to make sure it's foolproof with current usage. (Raising an error for that would probably break existing calls in the codebase that are otherwise working fine.)

Could probably test (maybe in separate PR?) by trying to raise an error, see if any e2e tests fail, then if so figuring out whether we should standardize all those cases and forbid future "query params in URLs", or allow such params but move them to the right places as needed in the interceptor.

Reply:
Gotcha. Created a new issue to track this: https://github.com/oppia/oppia/issues/14843.